sumpy/symbolic.py

```diff
@@ -153,48 +160,51 @@ def UnevaluatedExpr(x):  # noqa: N802
 
 
 if USE_SYMENGINE:
-    def doit(expr):
+    def doit(expr: Expr) -> Expr:
         return expr
 
-    def unevaluated_pow(a, b):
-        return sym.Pow(a, b)
+    def unevaluated_pow(a: Expr, b: complex | Expr) -> ArithmeticExpr:
+        return Pow(a, b)
 else:
-    def doit(expr):
+    def doit(expr: Expr) -> Expr:
         return expr.doit()
 
-    def unevaluated_pow(a, b):
-        return sym.Pow(a, b, evaluate=False)
+    def unevaluated_pow(a: Expr, b: complex | Expr) -> ArithmeticExpr:
+        return Pow(a, b, evaluate=False)
```

Comment:
This is most likely incorrect, but it seemed to be what `sympy.Pow` wanted? Not sure..

Reply:
What's incorrect about it? I'm not sure I'm seeing anything wrong.